### ISSUE TITLE
Migrations: Detect renamed entity types

### ISSUE THREAD

**Issue by ArunEdathadan** [dotnet/efcore#7337]

**Comment by divega** [comment]
**EF Team Triage:** This issue is lacking enough information for us to be able to effectively triage it. In particular, it is missing the following information requested in the new issue template. Can you please provide this information?

> ### Steps to reproduce
Ideally include a complete code listing that we can run to reproduce the issue.
Alternatively, you can provide a project/solution that we can run.

BTW we're not just doing this to be mean :smile:... we get a lot traffic on this project and it takes time to attempt to reproduce an issue based on fragments of information. In addition, our attempt is often unsuccessful as the exact conditions required to hit the issue are often not explicitly included in the code provided. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we ask that folks give us a self contained way to reproduce an issue.

For a guide on submitting good bug reports, read [Painless Bug Tracking](http://www.joelonsoftware.com/articles/fog0000000029.html).

**Comment by bricelam** [comment]
We don't currently detect entity type renames. We used to do this using fuzzy matching in EF6, but sometimes got it wrong. In EF Core, we decided that it's easier to rewrite dorp-add to `RenameTable` calls rather than sometimes having to change the rename to drop and add calls.

I think we could do something similar to #6582 here and detect them as a rename if *only* the name has changed and nothing else about the type.
